mediator/src/main/kotlin/pl/edu/prz/kod/mediator/domain/RunnerManager.kt

```diff
@@ -98,4 +100,23 @@ class RunnerManager(
         return response.status.successful && lenses.statusResponseLens(response).status == RunnerStatus.READY
     }
 
+    private fun persistResult(context: ExecutionContext, request: CodeRequest, result: ExecuteRequestResult) {
+        when (result) {
+            is ExecuteRequestResult.Success -> runBlocking {
+                logRepository.insertLog(
+                    ExecutionLog(
+                        email = context.email,
+                        language = request.language,
+                        code = request.base64Code,
+                        stdOut = result.codeResponse.stdOut,
+                        stdErr = result.codeResponse.stdErr,
+                        exitCode = result.codeResponse.exitCode
+                    )
+                )
+                logEvent(LogInsertedEvent(context.email))
+            }
+
+            else -> logEvent(FailedToInsertLogEvent(context.email))
```

Comment:
Is logging this event necessary? Its name would suggest that some error occured while trying to insert log into database, which is not true here. This event is logged if for whatever reason runner didn't successfully execute the request. And that's already logged as `ExecutionFailedEvent`. So I would say this line is not really needed here.